samples/3_map_state_simple/src/app/esri-map/esri-map.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { loadModules } from 'esri-loader';
+
+import { MapStateService } from '../services/map-state.service';
+
+@Component({
+  selector: 'app-esri-map',
+  templateUrl: './esri-map.component.html',
+  styleUrls: ['./esri-map.component.css']
+})
+
+export class EsriMapComponent implements OnInit {
+
+  public mapView: __esri.MapView;
+  msService: MapStateService;
+
+  // this is needed to be able to create the MapView at the DOM element in this component
+  @ViewChild('mapViewNode') private mapViewEl: ElementRef;
+
+  constructor(msService: MapStateService) {
+    this.msService = msService;
+  }
+
+  public ngOnInit() {
+    // use esri-loader to load JSAPI modules
+    return loadModules([
```

Comment:
I'd recommend leaving off the `return` and just run `loadModules()` like a normal method. I think I had that in my code but it was a left-over that I kept forgetting to remove.